tests/xmap_test.py

```diff
@@ -402,6 +403,43 @@ def testReductions(self, reduction, axes, mapped_axis):
     x = rng.randn(2, 5, 6)
     self.assertAllClose(ref_red(x), xmap_red(x))
 
+
+class NamedRandomTest(jtu.JaxTestCase):
+  def setUp(self):
+    if jax.lib.version < (0, 1, 58):
+      raise SkipTest("xmap requires jaxlib version >= 0.1.58")
+    if not config.omnistaging_enabled:
+      raise SkipTest("xmap requires omnistaging")
+
+  @parameterized.named_parameters(
+    {"testcase_name": name, "distr_sample": sample}
+    for name, sample in [
+      ("Uniform", jax.random.uniform),
+      ("Normal", jax.random.normal),
+      ("Bernoulli", partial(jax.random.bernoulli, p=0.5)),
+      ("TruncatedNormal", partial(jax.random.truncated_normal, lower=-2, upper=2)),
+    ])
+  @ignore_xmap_warning()
+  def testSample(self, distr_sample):
```

Comment:
It would be nice to test that the values we get out don't depend on the schedule, i.e. whether vectorized or parallelized. Maybe leave that as a TODO?

Reply:
This is a great idea! I thought it is important enough that I've added another commit that adds this test. Fortunately, my previous implementation worked ok!